be/src/storage/vectorized/schema_change.cpp

```diff
@@ -0,0 +1,1458 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "storage/vectorized/schema_change.h"
+
+#include <signal.h>
+#include <util/defer_op.h>
+
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+#include "runtime/exec_env.h"
+#include "runtime/heartbeat_flags.h"
+#include "runtime/mem_pool.h"
+#include "runtime/mem_tracker.h"
+#include "storage/merger.h"
+#include "storage/row.h"
+#include "storage/row_block.h"
+#include "storage/row_cursor.h"
+#include "storage/rowset/rowset_factory.h"
+#include "storage/rowset/rowset_id_generator.h"
+#include "storage/storage_engine.h"
+#include "storage/tablet.h"
+#include "storage/tablet_updates.h"
+#include "storage/vectorized/chunk_aggregator.h"
+#include "storage/vectorized/convert_helper.h"
+#include "storage/wrapper_field.h"
+#include "util/unaligned_access.h"
+
+using std::deque;
+using std::list;
+using std::nothrow;
+using std::pair;
+using std::string;
+using std::stringstream;
+using std::vector;
+
+namespace starrocks {
+namespace vectorized {
+
+using ChunkRow = std::pair<size_t, Chunk*>;
+
+int compare_chunk_row(const ChunkRow& lhs, const ChunkRow& rhs) {
+    for (uint16_t i = 0; i < lhs.second->schema()->num_key_fields(); ++i) {
+        int res = lhs.second->get_column_by_index(i)->compare_at(lhs.first, rhs.first,
+                                                                 *(rhs.second->get_column_by_index(i)), -1);
+        if (res != 0) {
+            return res;
+        }
+    }
+    return 0;
+}
+
+class ChunkSorter {
+public:
+    explicit ChunkSorter(ChunkAllocator* allocator);
+    virtual ~ChunkSorter();
+
+    bool sort(ChunkPtr& chunk, TabletSharedPtr new_tablet);
+    size_t allocated_rows() { return _max_allocated_rows; }
+
+    static bool _chunk_comparator(const ChunkRow& lhs, const ChunkRow& rhs) { return compare_chunk_row(lhs, rhs) < 0; }
+
+private:
+    ChunkAllocator* _chunk_allocator;
+    ChunkPtr _swap_chunk;
+    size_t _max_allocated_rows;
+};
+
+class ChunkMerger {
+public:
+    explicit ChunkMerger(MemTracker* mem_tracker, TabletSharedPtr tablet);
+    virtual ~ChunkMerger();
+
+    bool merge(std::vector<ChunkPtr>& chunk_arr, RowsetWriter* rowset_writer);
+    void aggregate_chunk(ChunkAggregator& aggregator, ChunkPtr& chunk, RowsetWriter* rowset_writer);
+
+private:
+    struct MergeElement {
+        bool operator<(const MergeElement& other) const {
+            return compare_chunk_row(std::make_pair(row_index, chunk), std::make_pair(other.row_index, other.chunk)) >
+                   0;
+        }
+
+        Chunk* chunk;
+        size_t row_index;
+    };
+
+    bool _make_heap(std::vector<ChunkPtr>& chunk_arr);
+    bool _pop_heap();
+
+    TabletSharedPtr _tablet;
+    std::unique_ptr<MemTracker> _mem_tracker = nullptr;
+    std::priority_queue<MergeElement> _heap;
+    ChunkAggregator* _aggregator;
```

Comment:
use std::unique_ptr to valid memory leak

Reply:
done